social_engagement/models.py

```diff
@@ -168,13 +168,37 @@ class StudentSocialEngagementScoreHistory(TimeStampedModel):
     score = models.IntegerField()
 
 
+class StudentSocialEngagementStats(TimeStampedModel):
+    """
+    Stores the social engagement stats for faster access and calculation.
+    """
+    score = models.OneToOneField(
+        StudentSocialEngagementScore,
+        on_delete=models.CASCADE,
+        related_name='stats',
+        primary_key=True
+    )
+    num_threads = models.IntegerField(default=0, db_index=True, null=False)
```

Comment:
All of the `null=False` here are redundant because it's [already the default](https://docs.djangoproject.com/en/1.8/ref/models/fields/#null).